spi/core-spi/src/test/java/org/eclipse/dataspaceconnector/spi/types/TypeManagerTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.spi.types;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.BeanSerializerFactory;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class TypeManagerTest {
+
+    @Test
+    void verifySerialization() throws JsonProcessingException {
+        var manager = new TypeManager();
+        manager.registerSerializer("foo", Bar.class, new JsonSerializer<>() {
```

Comment:
I'm late to the party but this test should be more strict as it succeed also if the serializer is not registered.